### PR TITLE
Feature/application patch

### PR DESCRIPTION
**Pull request by Jehfkemsy**
- Added application store
- Updated application  API to use `post` instead of `put` for `create`
- Fixed misspellings in application validation 
- Populated not required fields with `N/A` on submit
- Added `prettier` and `eslint` config so @aalexcomas11  can't ask me to fix my quotes again
- Removed `yarn.lock` and added it to `.gitignore`... ps. use `yarn` or `npm`, not both 😉

### CODE REVIEW

File: src/API/ApplicationAPI.js [hunk 1]
```diff
@@ -1,29 +1,38 @@
...
-      }
-    })
+        Authorization: getAuthToken(),
+      },
+    });
   },
-  getApplication: async(data)=> {
-    return await client.get('/application/get', {
+  getApplication: async (data) => {
+    return await client.get("/application/get", {
```
Comment (aalexcomas11) [inline src/API/ApplicationAPI.js:31]: single quotes, but i'll make an exception ;)

Comment (aalexcomas11) [inline src/API/ApplicationAPI.js:31]: also, I know it wasn't you, but can remove the await

File: src/store/ApplicationStore/index.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  return (initialValue) => {
+    const [value, set] = stateHook(initialValue);
+    return { value, set };
+  };
+};
+
+export default ({ children }) => {
+  const stateHook = createState(useState);
+
+  const user = stateHook(null);
```
Comment (aalexcomas11) [inline src/store/ApplicationStore/index.js:15]: How come the user is here? The user belongs in the auth store. This store should contain the application.

**Comment by Jehfkemsy** [comment]
@aalexcomas11  added requested changes... also updated pr description above

**Pull request merged**